src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/Serialization/CollectionTests.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text.Json.Serialization;
+using System.Text.Json.Serialization.Tests;
+
+namespace System.Text.Json.SourceGeneration.Tests
+{
+    public partial class CollectionTests_Metadata : CollectionTests
+    {
+        public CollectionTests_Metadata()
+            : this(new StringSerializerWrapper(CollectionTestsContext_Metadata.Default, (options) => new CollectionTestsContext_Metadata(options)))
+        {
+        }
+
+        protected CollectionTests_Metadata(JsonSerializerWrapperForString serializerWrapper)
+            : base(serializerWrapper, new StreamSerializerWrapper())
+        {
+        }
+
+        [JsonSourceGenerationOptions(GenerationMode = JsonSourceGenerationMode.Metadata)]
+        [JsonSerializable(typeof(ConcurrentDictionary<string, string>))]
+        [JsonSerializable(typeof(ConcurrentQueue<string>))]
+        [JsonSerializable(typeof(ConcurrentStack<string>))]
+        [JsonSerializable(typeof(BlockingCollection<string>))]
+        [JsonSerializable(typeof(ConcurrentBag<string>))]
+        [JsonSerializable(typeof(GenericConcurrentQueuePrivateConstructor<string>))]
+        [JsonSerializable(typeof(GenericConcurrentQueueInternalConstructor<string>))]
+        [JsonSerializable(typeof(GenericConcurrentStackPrivateConstructor<string>))]
+        [JsonSerializable(typeof(GenericConcurrentStackInternalConstructor<string>))]
+        [JsonSerializable(typeof(IDictionary))]
+        [JsonSerializable(typeof(Dictionary<string, string>))]
+        [JsonSerializable(typeof(SortedDictionary<string, string>))]
+        [JsonSerializable(typeof(object))]
+        [JsonSerializable(typeof(IDictionary<string, string>))]
+        [JsonSerializable(typeof(IReadOnlyDictionary<string, string>))]
+        [JsonSerializable(typeof(ImmutableDictionary<string, string>))]
+        [JsonSerializable(typeof(IImmutableDictionary<string, string>))]
+        [JsonSerializable(typeof(ImmutableSortedDictionary<string, string>))]
+        [JsonSerializable(typeof(Hashtable))]
+        [JsonSerializable(typeof(SortedList))]
+        [JsonSerializable(typeof(WrapperForIDictionary))]
+        //[JsonSerializable(typeof(StringToStringDictionaryWrapper))]
```

Comment:
Dead or future code? Here and below

Reply:
Future code, for the remaining collection tests. Will complete in next commit.